exif.cc

```diff
@@ -969,41 +939,38 @@ exif_put_value(const int ifd_nr, const uint16_t tag_id, const uint16_t type, con
       return nullptr;
     }
 
-    tag = (exif_tag_t*) xcalloc(sizeof(*tag), 1);
+    ifd->tags.append(ExifTag());
```

Comment:
It would be more natural to me as well, and I actually started down that path.  But in the end I went with less disruptive more literal translation.  Sometimes a pointer to a list element is returned and we need to get it back from the list anyway.